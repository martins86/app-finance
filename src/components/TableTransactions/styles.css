.table {
  max-width: 1120px;
  width: 92%;
  margin: 20px auto;
  display: flex;
  justify-content: space-around;
  background-color: #fff;
  border-radius: 5px;
  padding: 15px;
  box-shadow: 3px 2px 4px 1px #00000054;
  overflow-x: auto;
}

.table-container {
  width: 98%;
}

.table-titles th {
  padding: 8px;
  border-bottom: inset;
  text-align: start;
  width: 30%;
  height: 40px;
}

.table-titles th:first-child {
  width: 40%;
}

.table-titles th:nth-last-child(1) {
  text-align: center;
  width: 10%;
}

.table-titles th:nth-last-child(2) {
  text-align: center;
  width: 10%;
}
