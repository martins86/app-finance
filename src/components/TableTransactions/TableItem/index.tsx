import {
  FaRegArrowAltCircleDown,
  FaRegArrowAltCircleUp,
  FaTrash,
} from 'react-icons/fa';

import './styles.css';

export interface TableItemParams {
  item: any;
  onDelete: any;
}

const TableItem = ({ item, onDelete }: TableItemParams) => {
  return (
    <tr className="table-itens">
      <td>{item.desc}</td>
      <td>
        {Number(item.amount)?.toLocaleString('pt-br', {
          style: 'currency',
          currency: 'BRL',
        })}
      </td>
      <td>
        {item.expense ? (
          <FaRegArrowAltCircleDown title="SaÃ­da" color="red" />
        ) : (
          <FaRegArrowAltCircleUp title="Entrada" color="royalblue" />
        )}
      </td>
      <td>
        <FaTrash
          title="Remover item"
          onClick={() => onDelete(item.id)}
          color="gray"
        />
      </td>
    </tr>
  );
};

export default TableItem;
