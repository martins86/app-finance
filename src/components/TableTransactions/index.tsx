import './styles.css';

const TableTransactions = () => {
    return (
        <div>TableTransactions</div>
    );
};

export default TableTransactions;