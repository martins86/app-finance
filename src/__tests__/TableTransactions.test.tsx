import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import TableTransactions from '../components/FormTransactions';

describe('Test <TableTransactions />', () => {
  test('Should match snapshot', () => {
    // Arrange
    const fn = jest.fn();
    const transactionList = [''];
    const { container } = render(
      <Router>
        <TableTransactions
          transactionList={transactionList}
          setTransactionList={fn}
        />
      </Router>
    );

    // Act & Assert
    expect(container).toMatchSnapshot();
  });
});
