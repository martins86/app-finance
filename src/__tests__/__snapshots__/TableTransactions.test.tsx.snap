// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <TableTransactions /> Should match snapshot 1`] = `
<div>
  <div
    class="form-transactions"
  >
    <div
      class="form-fieldset"
    >
      <label
        for="desc"
      >
        Descrição
      </label>
      <input
        class="input-primary"
        id="desc"
        placeholder="Digite a descrição"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-fieldset"
    >
      <label
        for="amount"
      >
        Valor
      </label>
      <input
        class="input-primary"
        id="amount"
        min="1"
        placeholder="Digite o valor"
        type="number"
        value=""
      />
    </div>
    <div
      class="form-fieldset"
    >
      <div
        class="form-radio-group"
      >
        <div
          class="fieldset-item"
        >
          <input
            checked=""
            class="input-radio"
            id="rIncome"
            name="group1"
            type="radio"
            value=""
          />
          <label
            for="rIncome"
            title="Checar como entrada"
          >
            Entrada
          </label>
        </div>
        <div
          class="fieldset-item"
        >
          <input
            class="input-radio"
            id="rExpenses"
            name="group1"
            type="radio"
            value=""
          />
          <label
            for="rExpenses"
            title="Checar como saída"
          >
            Saída
          </label>
        </div>
      </div>
    </div>
    <div
      class="form-fieldset"
    >
      <button
        class="btn btn--primary"
        data-testid="button"
        title="Adicionar"
        type="button"
      >
        Adicionar
      </button>
    </div>
  </div>
  ,
</div>
`;
